In [12]:


import numpy as np


b = 1.50676807
w1 = -1.90038195
w2 = 1.35680768
x = 512 # max visits = 512
y = 512 # max visits = 512
R = 400 * (b + w1 * np.log(x) + w2 * np.log(y)) / np.log(10)
R # -327.3233531845416

-327.3233531845416

The expected ELO rating is modeled by the logistic regression:
$$ R = {400(b+w_1 \ln(x) + w_2 \ln(y)) \over {\ln(10)}} $$
where $b = 1.50676807$, $w_1 = -1.90038195$, and $w_2 = 1.35680768$. The $x$ and $y$ variables are max visits that are set in the KataGo configuration.

In [9]:
x = [32, 64, 128, 256]
y = np.exp(-(b + w1 * np.log(x)) / w2)
print(y)

[ 42.25319953 111.55536117 294.5244087  777.59263572]


To solve for $y$ when $R = 0$, we start with the equation:

$$ R = {400(b+w_1 \ln(x) + w_2 \ln(y)) \over {\ln(10)}} $$

This simplifies to:

$$ b + w_1 \ln(x) + w_2 \ln(y) = 0 $$

The solution for $y$ when $R = 0$ is confirmed to be:

$$ y = \exp\left(-\frac{b + w_1 \ln(x)}{w_2}\right) $$

| x   | y      |
|-----|--------|
| 32  | 42.25  |
| 64  | 111.56 |
| 128 | 294.52 |
| 256 | 777.59 |